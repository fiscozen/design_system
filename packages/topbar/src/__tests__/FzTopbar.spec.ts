import { describe, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { FzTopbar } from '..'

describe.concurrent('FzTopbar', () => {
  it('matches default topbar snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'default'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches danger topbar snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'danger'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches default topbar with button snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'default',
        style: 'button'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches danger topbar with button snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'danger',
        style: 'button'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches default topbar with icon-button snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'default',
        style: 'icon-button'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches danger topbar with icon-button snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'danger',
        style: 'icon-button'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches default topbar with link snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'default',
        style: 'link'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches danger topbar with link snapshot', async ({ expect }) => {
    const wrapper = mount(FzTopbar, {
      props: {
        type: 'danger',
        style: 'link'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
