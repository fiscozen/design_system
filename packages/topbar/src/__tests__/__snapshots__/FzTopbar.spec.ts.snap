// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzTopbar > matches danger topbar snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-danger"><span data-v-f59ba988=""> Testo qui </span>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`FzTopbar > matches danger topbar with button snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-danger"><span data-v-f59ba988=""> Testo qui </span><button data-v-f59ba988="" type="button" class="relative rounded flex flex-shrink items-center justify-center font-medium border-1 border-transparent h-28 text-sm focus:border-blue-600 focus:border-solid focus:border-1 px-14 text-core-white temporary-bg-red temporary-border-red bg-semantic-error">
    <!--v-if-->
    <!--v-if--><span class="hidden h-0 w-0"></span>
  </button>
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`FzTopbar > matches danger topbar with icon-button snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-danger"><span data-v-f59ba988=""> Testo qui </span>
  <!--v-if--><button data-v-f59ba988="" type="button" class="relative rounded flex flex-shrink items-center justify-center font-medium border-1 border-transparent h-28 text-sm focus:border-blue-600 focus:border-solid focus:border-1 px-14 text-grey-500 bg-transparent border-transparent hover:bg-grey-100 focus:!border-blue-600 disabled:text-grey-100 w-28 h-28">
    <!--v-if-->
    <div class="flex items-center justify-center w-[20px] h-[20px]">
      <!---->
    </div>
    <!--v-if-->
    <!--v-if--><span class="hidden h-0 w-0"></span>
  </button>
  <!--v-if-->
</div>"
`;

exports[`FzTopbar > matches danger topbar with link snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-danger"><span data-v-f59ba988=""> Testo qui </span>
  <!--v-if-->
  <!--v-if-->
  <router-link data-v-f59ba988="" replace="false" class="border-1 border-transparent text-sm hover:underline text-semantic-error hover:text-red-600 focus:text-red-600 focus:border-red-600"></router-link>
</div>"
`;

exports[`FzTopbar > matches default topbar snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-white-smoke"><span data-v-f59ba988=""> Testo qui </span>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`FzTopbar > matches default topbar with button snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-white-smoke"><span data-v-f59ba988=""> Testo qui </span><button data-v-f59ba988="" type="button" class="relative rounded flex flex-shrink items-center justify-center font-medium border-1 border-transparent h-28 text-sm focus:border-blue-600 focus:border-solid focus:border-1 px-14 bg-blue-500 hover:bg-blue-600 disabled:bg-blue-200 text-core-white focus:bg-blue-500">
    <!--v-if-->
    <!--v-if--><span class="hidden h-0 w-0"></span>
  </button>
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`FzTopbar > matches default topbar with icon-button snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-white-smoke"><span data-v-f59ba988=""> Testo qui </span>
  <!--v-if--><button data-v-f59ba988="" type="button" class="relative rounded flex flex-shrink items-center justify-center font-medium border-1 border-transparent h-28 text-sm focus:border-blue-600 focus:border-solid focus:border-1 px-14 text-grey-500 bg-transparent border-transparent hover:bg-grey-100 focus:!border-blue-600 disabled:text-grey-100 w-28 h-28">
    <!--v-if-->
    <div class="flex items-center justify-center w-[20px] h-[20px]">
      <!---->
    </div>
    <!--v-if-->
    <!--v-if--><span class="hidden h-0 w-0"></span>
  </button>
  <!--v-if-->
</div>"
`;

exports[`FzTopbar > matches default topbar with link snapshot 1`] = `
"<div data-v-f59ba988="" class="flex px-24 py-12 gap-16 items-center justify-center bg-white-smoke"><span data-v-f59ba988=""> Testo qui </span>
  <!--v-if-->
  <!--v-if-->
  <router-link data-v-f59ba988="" replace="false" class="border-1 border-transparent text-sm hover:underline text-blue-500 hover:text-blue-600 focus:text-blue-600 focus:border-blue-600"></router-link>
</div>"
`;
